clippy_lints/src/manual_assert.rs

```diff
@@ -47,6 +47,7 @@ impl<'tcx> LateLintPass<'tcx> for ManualAssert {
             if cx.tcx.item_name(macro_call.def_id) == sym::panic;
             if !cx.tcx.sess.source_map().is_multiline(cond.span);
             if let Some(format_args) = FormatArgsExpn::find_nested(cx, then, macro_call.expn);
+            if !is_else_clause(cx.tcx, expr);
```

Comment:
I don't think the suggestion is wrong. Couldn't we just wrap the suggestion in `{ ... }` in the `else_clause` case?

Reply:
That would also work, but would probably annoy people who would otherwise like the lint (me being one of them). It could be a config to enable linting a trailing `else if`. 

Linting here doesn't have the same benefit of removing the whole if expression the other cases do. Changing
```rust
} else if foo {
    panic!("oh no!");
}
```
to
```rust
} else {
    assert!(foo, "oh no!");
}
```
is an even more pedantic change than the regular case. Because the `else` clause still needs to exist, all this accomplishes is shuffling the condition around, rather than remove a construct from the code.

Reply:
Yeah, I kind of agree with jarcho here.

However, I would like a comment here stating why we skip the else clause